src/Microsoft.Health.Fhir.Core/Features/Validation/ISupportedProfilesStore.cs

```diff
@@ -14,7 +15,7 @@ public interface ISupportedProfilesStore
         /// </summary>
         /// <param name="resourceType">Resource type to get profiles.</param>
         /// <param name="disableCacheRefresh">Should we check server for new updates or get data out of cache.</param>
-        IEnumerable<string> GetSupportedProfiles(string resourceType, bool disableCacheRefresh = false);
+        Task<IEnumerable<string>> GetSupportedProfiles(string resourceType, bool disableCacheRefresh = false);
```

Comment:
Is it possible to rename this method to GetSupportedProfilesAsync and expect a cancellation token?